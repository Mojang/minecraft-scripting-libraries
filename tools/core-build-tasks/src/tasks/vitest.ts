import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
