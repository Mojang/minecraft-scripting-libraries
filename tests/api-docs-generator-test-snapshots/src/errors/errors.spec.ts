// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Errors', () => {
    it('Generates correct output for error objects', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['ts', 'msdocs'],
        });
    });
});
