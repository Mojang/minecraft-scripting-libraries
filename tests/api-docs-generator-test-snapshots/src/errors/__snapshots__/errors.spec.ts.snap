// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Errors > Generates correct output for error objects > # Of Files Generated 1`] = `10`;

exports[`Errors > Generates correct output for error objects > msdocs/scriptapi/TOC.yml 1`] = `
"  - name: minecraft/ExampleModule
    items:
    - name: minecraft/ExampleModule
      href: minecraft/examplemodule/minecraft-examplemodule.md
    - name: ExampleClass
      href: minecraft/examplemodule/ExampleClass.md
  - name: minecraft/ExampleModule2
    items:
    - name: minecraft/ExampleModule2
      href: minecraft/examplemodule2/minecraft-examplemodule2.md
    - name: CrossModuleReturn
      href: minecraft/examplemodule2/CrossModuleReturn.md
"
`;

exports[`Errors > Generates correct output for error objects > msdocs/scriptapi/minecraft/examplemodule/ExampleClass.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: minecraft/examplemodule.ExampleClass Class
description: Contents of the minecraft/ExampleModule.ExampleClass class.
---
# ExampleClass Class

## Properties

### **examplePropertyThatThrows**
\`examplePropertyThatThrows: number;\`

Type: *number*

Notes:
  - This property can't be used in read-only mode.
  - This property can throw errors when used.

### **examplePropertyThatThrowsCustomError**
\`examplePropertyThatThrowsCustomError: number;\`

Type: *number*

Notes:
  - This property can't be used in read-only mode.
  - This property can throw errors when used.
    - Throws [*ExampleError*](ExampleError.md)

### **examplePropertyThatThrowsTwoErrors**
\`examplePropertyThatThrowsTwoErrors: number;\`

Type: *number*

Notes:
  - This property can't be used in read-only mode.
  - This property can throw errors when used.
    - Throws [*ExampleError*](ExampleError.md), [*ExampleError2*](ExampleError2.md)

### **examplePropertyThatThrowsWithReadOnlyPrivilege**
\`examplePropertyThatThrowsWithReadOnlyPrivilege: number;\`

Type: *number*

Notes:
  - This property can throw errors when used.
    - Throws [*ExampleError*](ExampleError.md)

### **examplePropertyWithCrossModuleError**
\`examplePropertyWithCrossModuleError: number;\`

Type: *number*

Notes:
  - This property can't be used in read-only mode.
  - This property can throw errors when used.
    - Throws [*minecraft/ExampleModule2.CrossModuleExampleError*](../../../scriptapi/minecraft/examplemodule2/CrossModuleExampleError.md)

### **examplePropertyWithCrossModuleReturnAndError**
\`examplePropertyWithCrossModuleReturnAndError: minecraftexamplemodule2.CrossModuleReturn;\`

Type: *minecraft/ExampleModule2.CrossModuleReturn*

Notes:
  - This property can't be used in read-only mode.
  - This property can throw errors when used.
    - Throws [*ExampleError*](ExampleError.md)

## Methods
- [exampleMethodThatThrows](#examplemethodthatthrows)
- [exampleMethodThatThrowsCustomError](#examplemethodthatthrowscustomerror)
- [exampleMethodThatThrowsTwoErrors](#examplemethodthatthrowstwoerrors)
- [exampleMethodThatThrowsWithReadOnlyPrivilege](#examplemethodthatthrowswithreadonlyprivilege)

### **exampleMethodThatThrows**
\`
exampleMethodThatThrows(): string
\`

**Returns** *string*
  
Notes:
- This function can't be called in read-only mode.
- This function can throw errors.

### **exampleMethodThatThrowsCustomError**
\`
exampleMethodThatThrowsCustomError(): string
\`

**Returns** *string*
  
Notes:
- This function can't be called in read-only mode.
- This function can throw errors.
  - Throws [*ExampleError*](ExampleError.md)

### **exampleMethodThatThrowsTwoErrors**
\`
exampleMethodThatThrowsTwoErrors(): string
\`

**Returns** *string*
  
Notes:
- This function can't be called in read-only mode.
- This function can throw errors.
  - Throws [*ExampleError*](ExampleError.md), [*ExampleError2*](ExampleError2.md)

### **exampleMethodThatThrowsWithReadOnlyPrivilege**
\`
exampleMethodThatThrowsWithReadOnlyPrivilege(): string
\`

**Returns** *string*
  
Notes:
- This function can throw errors.
  - Throws [*ExampleError*](ExampleError.md)
"
`;

exports[`Errors > Generates correct output for error objects > msdocs/scriptapi/minecraft/examplemodule/ExampleError.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: minecraft/examplemodule.ExampleError Class
description: Contents of the minecraft/ExampleModule.ExampleError class.
---
# ExampleError Class

## Extends
- *Error*

## Properties

### **exampleOptionalProperty**
\`read-only exampleOptionalProperty?: string;\`

Type: *string*

### **exampleProperty**
\`read-only exampleProperty: string[];\`

Type: *string*[]
"
`;

exports[`Errors > Generates correct output for error objects > msdocs/scriptapi/minecraft/examplemodule/ExampleError2.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: minecraft/examplemodule.ExampleError2 Class
description: Contents of the minecraft/ExampleModule.ExampleError2 class.
---
# ExampleError2 Class

## Extends
- *Error*

## Properties

### **exampleProperty**
\`read-only exampleProperty: string[];\`

Type: *string*[]
"
`;

exports[`Errors > Generates correct output for error objects > msdocs/scriptapi/minecraft/examplemodule/minecraft-examplemodule.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: minecraft/examplemodule Module
description: Contents of the minecraft/ExampleModule module
---
# \`minecraft/ExampleModule\` Module

## Manifest Details
\`\`\`json
{
    "module_name": "minecraft/ExampleModule",
    "version": "1.0.0"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`1.0.0\`

## Classes
- [ExampleClass](ExampleClass.md)

## Errors
- [ExampleError](ExampleError.md)
- [ExampleError2](ExampleError2.md)

## Functions

### **exampleFunctionThatThrows**
\`
exampleFunctionThatThrows(): string
\`

**Returns** *string*
  
Notes:
- This function can't be called in read-only mode.
- This function can throw errors.

### **exampleFunctionThatThrowsCustomError**
\`
exampleFunctionThatThrowsCustomError(): string
\`

**Returns** *string*
  
Notes:
- This function can't be called in read-only mode.
- This function can throw errors.
  - Throws [*ExampleError*](ExampleError.md)

### **exampleFunctionThatThrowsTwoErrors**
\`
exampleFunctionThatThrowsTwoErrors(): string
\`

**Returns** *string*
  
Notes:
- This function can't be called in read-only mode.
- This function can throw errors.
  - Throws [*ExampleError*](ExampleError.md), [*ExampleError2*](ExampleError2.md)

### **exampleFunctionWithCrossModuleError**
\`
exampleFunctionWithCrossModuleError(): number
\`

**Returns** *number*
  
Notes:
- This function can't be called in read-only mode.
- This function can throw errors.
  - Throws [*minecraft/ExampleModule2.CrossModuleExampleError*](../../../scriptapi/minecraft/examplemodule2/CrossModuleExampleError.md)

### **exampleFunctionWithCrossModuleReturnAndError**
\`
exampleFunctionWithCrossModuleReturnAndError(): minecraftexamplemodule2.CrossModuleReturn
\`

**Returns** *minecraft/ExampleModule2.CrossModuleReturn*
  
Notes:
- This function can't be called in read-only mode.
- This function can throw errors.
  - Throws [*ExampleError*](ExampleError.md)

## Dependencies
- [\`minecraft/ExampleModule2\`](../../../scriptapi/minecraft/examplemodule2/minecraft-examplemodule2.md)
"
`;

exports[`Errors > Generates correct output for error objects > msdocs/scriptapi/minecraft/examplemodule2/CrossModuleExampleError.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: minecraft/examplemodule2.CrossModuleExampleError Class
description: Contents of the minecraft/ExampleModule2.CrossModuleExampleError class.
---
# CrossModuleExampleError Class

## Extends
- *Error*
"
`;

exports[`Errors > Generates correct output for error objects > msdocs/scriptapi/minecraft/examplemodule2/CrossModuleReturn.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: minecraft/examplemodule2.CrossModuleReturn Class
description: Contents of the minecraft/ExampleModule2.CrossModuleReturn class.
---
# CrossModuleReturn Class
"
`;

exports[`Errors > Generates correct output for error objects > msdocs/scriptapi/minecraft/examplemodule2/minecraft-examplemodule2.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: minecraft/examplemodule2 Module
description: Contents of the minecraft/ExampleModule2 module
---
# \`minecraft/ExampleModule2\` Module

## Manifest Details
\`\`\`json
{
    "module_name": "minecraft/ExampleModule2",
    "version": "1.0.0"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`1.0.0\`

## Classes
- [CrossModuleReturn](CrossModuleReturn.md)

## Errors
- [CrossModuleExampleError](CrossModuleExampleError.md)
"
`;

exports[`Errors > Generates correct output for error objects > typescript/minecraft/ExampleModule@1.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "minecraft/ExampleModule",
 *   "version": "1.0.0"
 * }
 * \`\`\`
 *
 */
import * as minecraftexamplemodule2 from 'minecraft/ExampleModule2';
export class ExampleClass {
    private constructor();
    /**
     * @remarks
     * This property can't be used in read-only mode.
     *
     * @throws This property can throw when used.
     */
    examplePropertyThatThrows: number;
    /**
     * @remarks
     * This property can't be used in read-only mode.
     *
     * @throws This property can throw when used.
     *
     * {@link ExampleError}
     */
    examplePropertyThatThrowsCustomError: number;
    /**
     * @remarks
     * This property can't be used in read-only mode.
     *
     * @throws This property can throw when used.
     *
     * {@link ExampleError}
     *
     * {@link ExampleError2}
     */
    examplePropertyThatThrowsTwoErrors: number;
    /**
     * @throws This property can throw when used.
     *
     * {@link ExampleError}
     */
    examplePropertyThatThrowsWithReadOnlyPrivilege: number;
    /**
     * @remarks
     * This property can't be used in read-only mode.
     *
     * @throws This property can throw when used.
     *
     * {@link minecraftexamplemodule2.CrossModuleExampleError}
     */
    examplePropertyWithCrossModuleError: number;
    /**
     * @remarks
     * This property can't be used in read-only mode.
     *
     * @throws This property can throw when used.
     *
     * {@link ExampleError}
     */
    examplePropertyWithCrossModuleReturnAndError: minecraftexamplemodule2.CrossModuleReturn;
    /**
     * @remarks
     * This function can't be called in read-only mode.
     *
     * @throws This function can throw errors.
     */
    exampleMethodThatThrows(): string;
    /**
     * @remarks
     * This function can't be called in read-only mode.
     *
     * @throws This function can throw errors.
     *
     * {@link ExampleError}
     */
    exampleMethodThatThrowsCustomError(): string;
    /**
     * @remarks
     * This function can't be called in read-only mode.
     *
     * @throws This function can throw errors.
     *
     * {@link ExampleError}
     *
     * {@link ExampleError2}
     */
    exampleMethodThatThrowsTwoErrors(): string;
    /**
     * @throws This function can throw errors.
     *
     * {@link ExampleError}
     */
    exampleMethodThatThrowsWithReadOnlyPrivilege(): string;
}

// @ts-ignore Class inheritance allowed for native defined classes
export class ExampleError extends Error {
    private constructor();
    exampleOptionalProperty?: string;
    exampleProperty: string[];
}

// @ts-ignore Class inheritance allowed for native defined classes
export class ExampleError2 extends Error {
    private constructor();
    exampleProperty: string[];
}

/**
 * @remarks
 * This function can't be called in read-only mode.
 *
 * @throws This function can throw errors.
 */
export function exampleFunctionThatThrows(): string;
/**
 * @remarks
 * This function can't be called in read-only mode.
 *
 * @throws This function can throw errors.
 *
 * {@link ExampleError}
 */
export function exampleFunctionThatThrowsCustomError(): string;
/**
 * @remarks
 * This function can't be called in read-only mode.
 *
 * @throws This function can throw errors.
 *
 * {@link ExampleError}
 *
 * {@link ExampleError2}
 */
export function exampleFunctionThatThrowsTwoErrors(): string;
/**
 * @remarks
 * This function can't be called in read-only mode.
 *
 * @throws This function can throw errors.
 *
 * {@link minecraftexamplemodule2.CrossModuleExampleError}
 */
export function exampleFunctionWithCrossModuleError(): number;
/**
 * @remarks
 * This function can't be called in read-only mode.
 *
 * @throws This function can throw errors.
 *
 * {@link ExampleError}
 */
export function exampleFunctionWithCrossModuleReturnAndError(): minecraftexamplemodule2.CrossModuleReturn;
"
`;

exports[`Errors > Generates correct output for error objects > typescript/minecraft/ExampleModule2@1.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "minecraft/ExampleModule2",
 *   "version": "1.0.0"
 * }
 * \`\`\`
 *
 */
export class CrossModuleReturn {
    private constructor();
}

// @ts-ignore Class inheritance allowed for native defined classes
export class CrossModuleExampleError extends Error {
    private constructor();
}
"
`;
